lib/grape/validations/validators/coerce.rb

```diff
@@ -54,7 +54,14 @@ def coerce_value(type, val)
         return val || Set.new if type == Set
         return val || {}      if type == Hash
 
-        converter = Virtus::Attribute.build(type)
+        # To support custom types that Virtus can't easily coerce, pass in an
+        # explicit coercer. Custom types must implement a `parse` class method.
+        converter_options = {}
+        if ![DateTime, Date, Time].include?(type) && type.respond_to?(:parse)
```

Comment:
Also if Date types are potentially special, make sure to write a test around them explicitly.